src/meta/src/hummock/manager/mod.rs

```diff
@@ -897,6 +897,51 @@ where
         Ok(())
     }
 
+    fn is_compact_task_expired(compact_task: &CompactTask, levels: &Levels) -> bool {
+        for input_level in compact_task.get_input_ssts() {
+            let info_map: HashMap<_, _> = match input_level.get_level_idx() {
+                0 => match levels.l0 {
```

Comment:
There may be several input-level zero in compact-task, so we may construct the same hash-map for several times. And I think there may be a lot of small files in level0, maybe we can store them as a hash-map in somewhere for every request.

Reply:
> There may be several input-level zero in compact-task, so we may construct the same hash-map for several times. And I think there may be a lot of small files in level0, maybe we can store them as a hash-map in somewhere for every request.

fixed.